### DISCUSSION TITLE
Documentation query ...  oauth-on-office-365  full_access_as_app

### DISCUSSION THREAD

**Discussion by williamhamilton** [ecederstrand/exchangelib#1175] [Q&A]
I am having trouble migrating an old script to use OAUTH on O365.  The documentation says "Continue to the API permissions page, and add the full_access_as_app permission" but I have been unable to find this as Exchange Online and full_access_as_app permission is not an option. 
Documentation I am looking at:  https://ecederstrand.github.io/exchangelib/#oauth-on-office-365

Is there example authentication code with token refresh that I can test against my configuration? I would like to work through permissions by using some known good code so that my code is not the cause of any issues.

**Comment by ecederstrand** [comment]
I think this question is better directed to an EWS or Azure-specific forum. I don't have a lot of experience with Azure and the permissions setup there. I just dumped whatever worked for me at the time in the docs. All I know is that the OAuth token you use for authentication must have access to the EWS API.

If you do find a solution to your problem, please post it here so we can update the docs.

**Comment by williamhamilton** [comment]
Thanks for responding so quickly..  I will continue reading the Microsoft docs and report back if/when I get it working.
